### PR TITLE
fix insert error catch, add insert copy over 

### LINKED ISSUE #47: add date/datetime/time conversion for copy over feature in insert mode
convert the select tuple's date/datetime/time values correctly so it shows up on the insert field and inserts correctly when copied over.

### LINKED ISSUE #71: implement proper error catch for insert
duplicate primary key insertion error breaks the insertion process - error catch needs to be moved up.

### LINKED ISSUE #82: Enhance 'remove filter card' operation
Replace the `Delete` button with a garbage bin button on lower left to be more clear that it is to remove the card only.

### PR DESCRIPTION
**Pull request by mahos**
- add catch error for insert errors - fixes #71 
Screen Shot 2021-02-25 at 4 37 39 PM

- make sure date, datetime, time, and timestamp fields get copied over in insert-copy - resolves #47
Screen Shot 2021-02-25 at 4 36 59 PM
Screen Shot 2021-02-25 at 4 37 16 PM

- replace word delete with trash icon for filter card - resolves #82
Screen Shot 2021-02-25 at 4 39 34 PM

### CODE REVIEW

File: src/Components/MainTableView/UpdateTuple/UpdateTuple.tsx [hunk 1]
```diff
@@ -71,13 +71,83 @@ class UpdateTuple extends React.Component<{token: string, selectedSchemaName:str
...
+      // let tupleBuffer = Object.assign({}, this.state.tupleBuffer);
+      // Object.values(this.props.tupleToUpdate).forEach((columns: any) => {
+      //   Object.entries(columns.primaryEntries).forEach((attributeKeyVal: any) => {
+      //     tupleBuffer[attributeKeyVal[0]] = attributeKeyVal[1]
+      //   })
+      //   Object.entries(columns.secondaryEntries).forEach((attributeKeyVal: any) => {
+      //     tupleBuffer[attributeKeyVal[0]] = attributeKeyVal[1]
+      //   })
+      // })
+      // this.setState({tupleBuffer: tupleBuffer});
```
Comment (guzman-raphael) [inline src/Components/MainTableView/UpdateTuple/UpdateTuple.tsx:83]: @mahos Was this intentionally left behind?

**Review by guzman-raphael — CHANGES_REQUESTED** [review]
Thanks, @mahos. Could you add a screenshot/GIF demonstrating the visual changes?

Specifically:
- What happens now when trying to insert a record with a duplicate primary key
- Copy over for a the newly supported types

File: src/Components/MainTableView/DataStorageClasses/TableAttribute.tsx [hunk 2]
```diff
@@ -280,35 +280,54 @@ class TableAttribute {
...
+        }
+        else {
+          return(
+            <select defaultValue={defaultValue} onChange={(e) => handleChange(e, tableAttribute.attributeName)}> {
+              tableAttribute.enumOptions?.map((enumOptionString: string) => {
+                return(<option selected={currentValue === enumOptionString} key={enumOptionString} value={enumOptionString}>{enumOptionString}</option>);
+              })}
+            </select>
+        )
+        }
```
Comment (Synicix) [inline src/Components/MainTableView/DataStorageClasses/TableAttribute.tsx:324]: Uh why is the branch needed here, also I don't think this handle the issuewith no default value correctly

File: src/Components/MainTableView/UpdateTuple/UpdateTuple.tsx (hunk 1 above)
Comment (mahos) [reply src/Components/MainTableView/UpdateTuple/UpdateTuple.tsx:83]: oh good catch - no - I meant to remove once the new conversion worked. Will remove

File: src/Components/MainTableView/InsertTuple/InsertTuple.tsx [hunk 3]
```diff
@@ -75,13 +75,73 @@ class InsertTuple extends React.Component<{token: string, selectedSchemaName:str
    */
   copyTuple(event: any, tupleToInsert: any) {
     event.preventDefault();
-    let tupleBuffer = Object.assign({}, this.state.tupleBuffer);
+    let tupleBuffer: any = {};
     Object.values(tupleToInsert).forEach((columns: any) => {
-      Object.entries(columns.primaryEntries).forEach((attributeKeyVal: any) => {
-        tupleBuffer[attributeKeyVal[0]] = attributeKeyVal[1]
+      Object.values(columns.tableAttributesInfo).forEach((attributes: any) => {
```
Comment (Synicix) [inline src/Components/MainTableView/InsertTuple/InsertTuple.tsx:80]: Isn't columns = tuples here, unless the naming is off...

File: src/Components/MainTableView/InsertTuple/InsertTuple.tsx [hunk 4]
```diff
@@ -75,13 +75,73 @@ class InsertTuple extends React.Component<{token: string, selectedSchemaName:str
    */
```
Comment (Synicix) [inline src/Components/MainTableView/InsertTuple/InsertTuple.tsx:75]: Can you update docks to be more detail what is tupleToInsert is exactly, also the TODO stuff is outdated for this pull request

File: src/Components/MainTableView/UpdateTuple/UpdateTuple.tsx [hunk 5]
```diff
@@ -71,13 +71,73 @@ class UpdateTuple extends React.Component<{token: string, selectedSchemaName:str
...
+                  }
+                })
+              }
+            }
+          })
+        });
+        Object.entries(columns.primaryEntries).forEach((attributeKeyVal: Array<any>) => {
+          if (!tupleBuffer[attributeKeyVal[0]]) {
+            tupleBuffer[attributeKeyVal[0]] = attributeKeyVal[1]
+          }
```
Comment (Synicix) [inline src/Components/MainTableView/UpdateTuple/UpdateTuple.tsx:135]: Okay yeah I am not letting this fly this time, this code seems to be a copy of code in the insert function.

**Review by Synicix — CHANGES_REQUESTED** [review]

File: src/Components/MainTableView/UpdateTuple/UpdateTuple.tsx (hunk 5 above)
Comment (mahos) [reply src/Components/MainTableView/UpdateTuple/UpdateTuple.tsx]: we worked hard on this.

**Review by Synicix — APPROVED** [review]
Worked on with Maho, thus approved, unless I miss something

**Comment by guzman-raphael** [comment]
Unlinked #83 since partially fixed. @Synicix will be working on fix for update side.

**Review by guzman-raphael — APPROVED** [review]

**Pull request merged**